tidy3d/components/geometry.py

```diff
@@ -3105,8 +3108,149 @@ def _surface_area(self, bounds: Bound) -> float:
+            axis=np.arange(3),
+        )
+        vertices = SurfaceMeshDataArray(vectors, coords=coords)
+        mesh_dataset = SurfaceMeshDataset(vertices=vertices)
+        return CustomSurfaceMeshGeometry(mesh_dataset=mesh_dataset)
+
+    @cached_property
+    def _normals(self) -> np.ndarray:
+        """Unnormalized outward-facing normals of each triangle.
+        The length of each normal equals the area of the triangle."""
+        vertices = self.vectors
+        vertex0 = vertices[:, 0, :]
+        vertex1 = vertices[:, 1, :]
+        vertex2 = vertices[:, 2, :]
+        return 0.5 * np.cross(vertex1 - vertex0, vertex2 - vertex0)
+
+    @cached_property
+    def _unit_normals(self) -> np.ndarray:
+        """Unit outward-facing normals of each triangle."""
+        normals = self._normals
+        return normals / np.linalg.norm(normals, axis=1)[:, None]
+
+    @cached_property
+    def _areas(self) -> ArrayLike[1]:
+        """Areas of each triangle"""
+        return np.linalg.norm(self._normals, axis=1)
+
+    def _surface_area(self, bounds: Bound) -> float:
+        normals = self._normals
+        return np.sum(np.linalg.norm(normals, axis=1))
+
+    def _volume(self, bounds: Bound) -> float:
+        normals_x = self._normals[:, 0]
+        volume_contributions = normals_x * np.sum(self.vectors[:, :, 0], axis=1) / 3
+        return np.sum(volume_contributions)
+
+    @cached_property
+    def bounds(self) -> Bound:  # pylint:disable=too-many-locals
+        if self.mesh_dataset is None:
+            return ((-inf, -inf, -inf), (inf, inf, inf))
+        bmin = np.amin(self.vectors, axis=(0, 2))
+        bmax = np.amax(self.vectors, axis=(0, 2))
+        return (bmin, bmax)
+
+    def intersections(self, x: float = None, y: float = None, z: float = None) -> List[Shapely]:
+        if self.mesh_dataset is None:
+            return []
+
+        axis, position = self.parse_xyz_kwargs(x=x, y=y, z=z)
+
+        in_plane_coords = [0, 1, 2]
+        in_plane_coords.pop(axis)
+        segments = []
+        for triangle in self.vectors:
+            points = []
+            for vertex_inds in [[1, 2], [0, 2], [0, 1]]:
+                vertex0 = triangle[vertex_inds[0]]
+                vertex1 = triangle[vertex_inds[1]]
+                if abs(vertex0[axis] - vertex1[axis]) < fp_eps:
+                    # TODO: handle this elsewhere (zero area)
+                    continue
+                time = (position - vertex1[axis]) / (vertex0[axis] - vertex1[axis])
+                if 0 <= time <= 1:
+                    point3d = vertex0 * time + vertex1 * (1 - time)
+                    points.append(point3d[in_plane_coords])
+            if len(points) == 1:
+                segments.append([points[0], points[0]])
+            elif len(points) == 2:
+                segments.append(points)
+            elif len(points) == 3:
+                # TODO: this case shouldn't happen unless triangles have zero area
+                pass
```

Comment:
probably dont want to `pass` here, better to raise an error and tell the user to make an issue on the front end if this comes up.